code-challenges.test.js

```diff
@@ -24,10 +24,37 @@ var secretCodeWord2 = "Gobbledygook"
 var secretCodeWord3 = "Eccentric"
 // Expected output: "3cc3ntr1c"
 
+describe("Coding Message Function", () => {
+    it("Takes a string and returns a coded message", () => {
+        let expected1 = "L4ck4d41s1c4l"
+        let expected2 = "G0bbl3dyg00k"
+        let expected3 = "3cc3ntr1c"
+
+        expect(encodeWord(secretCodeWord1)).toEqual(expected1)
+        expect(encodeWord(secretCodeWord2)).toEqual(expected2)
+        expect(encodeWord(secretCodeWord3)).toEqual(expected3)
+    })
+})
 
 
 // b) Create the function that makes the test pass.
 
+/*
+encodeWord
+param word - string
+convert the letters (both upper and lowercase!):
+    a, A -> 4, e, E -> 3, i, I -> 1, o, O -> 0
+use str.replaceAll(substr, newStr) instead of str.replace(),
+because .replace() will only replace first occurrence
+return the word
+*/
+
+const encodeWord = (word) => {
+    return word.replaceAll('a', '4').replaceAll('A', '4')
+                .replaceAll('e', '3').replaceAll('E', '3')
+                .replaceAll('i', '1').replaceAll('I', '1')
+                .replaceAll('o', '0').replaceAll('O', '0')
+}
```

Comment:
appended dot notation in JS can be quite handy sometimes. Great job keeping it all legible!